src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntTests.cs

```diff
@@ -1465,16 +1465,42 @@ static void getArgs(ArrayBuilder<object[]> builder, string op, string opType, st
 
             getArgs(builder, "+", "nint", "nint nint.op_UnaryPlus(nint value)");
             getArgs(builder, "+", "nuint", "nuint nuint.op_UnaryPlus(nuint value)");
-            getArgs(builder, "-", "nint", "nint nint.op_UnaryMinus(nint value)");
-            getArgs(builder, "-", "nuint", null);
-            getArgs(builder, "~", "nint", "nint nint.op_UnaryNot(nint value)");
-            getArgs(builder, "~", "nuint", "nuint nuint.op_UnaryNot(nuint value)");
+            getArgs(builder, "+", "System.IntPtr");
+            getArgs(builder, "+", "System.UIntPtr");
+            getArgs(builder, "-", "nint", "nint nint.op_UnaryNegation(nint value)");
+            getArgs(builder, "-", "nuint", null, Diagnostic(ErrorCode.ERR_AmbigUnaryOp, "-operand").WithArguments("-", "nuint")); // PROTOTYPE: Should report ERR_BadUnaryOp.
+            getArgs(builder, "-", "System.IntPtr");
+            getArgs(builder, "-", "System.UIntPtr");
+            getArgs(builder, "!", "nint");
+            getArgs(builder, "!", "nuint");
+            getArgs(builder, "!", "System.IntPtr");
+            getArgs(builder, "!", "System.UIntPtr");
+            getArgs(builder, "~", "nint", "nint nint.op_OnesComplement(nint value)");
+            getArgs(builder, "~", "nuint", "nuint nuint.op_OnesComplement(nuint value)");
+            getArgs(builder, "~", "System.IntPtr");
+            getArgs(builder, "~", "System.UIntPtr");
+
+            getArgs(builder, "+", "nint?", "nint nint.op_UnaryPlus(nint value)");
+            getArgs(builder, "+", "nuint?", "nuint nuint.op_UnaryPlus(nuint value)");
+            getArgs(builder, "+", "System.IntPtr?");
+            getArgs(builder, "+", "System.UIntPtr?");
+            getArgs(builder, "-", "nint?", "nint nint.op_UnaryNegation(nint value)");
+            getArgs(builder, "-", "nuint?", null, Diagnostic(ErrorCode.ERR_AmbigUnaryOp, "-operand").WithArguments("-", "nuint?")); // PROTOTYPE: Should report ERR_BadUnaryOp.
+            getArgs(builder, "-", "System.IntPtr?");
+            getArgs(builder, "-", "System.UIntPtr?");
+            getArgs(builder, "!", "nint?");
+            getArgs(builder, "!", "nuint?");
+            getArgs(builder, "!", "System.IntPtr?");
+            getArgs(builder, "!", "System.UIntPtr?");
+            getArgs(builder, "~", "nint?", "nint nint.op_OnesComplement(nint value)");
+            getArgs(builder, "~", "nuint?", "nuint nuint.op_OnesComplement(nuint value)");
+            getArgs(builder, "~", "System.IntPtr?");
+            getArgs(builder, "~", "System.UIntPtr?");
 
-            // PROTOTYPE: Test nint? and nuint?
             return builder;
         }
 
-        [Theory(Skip = "PROTOTYPE")]
+        [Theory]
         [MemberData(nameof(UnaryOperatorsData))]
         public void UnaryOperators(string op, string opType, string expectedSymbol, DiagnosticDescription[] expectedDiagnostics)
```

Comment:
>UnaryOperators [](start = 20, length = 14)

I think for each supported unary operator we should have a test that verifies the IL we produce and expected behavior at runtime. Should probably test behavior for lifted forms as well.